src/dns_encode.erl

```diff
@@ -1172,6 +1172,16 @@ do_encode_optrrdata(
 ) ->
     Data = <<FAMILY:16, SRCPL:8, SCOPEPL:8, ADDRESS/binary>>,
     {?DNS_EOPTCODE_ECS, Data};
+do_encode_optrrdata(#dns_opt_cookie{client = <<ClientCookie:8/binary>>, server = undefined}) ->
+    {?DNS_EOPTCODE_COOKIE, ClientCookie};
+do_encode_optrrdata(#dns_opt_cookie{
+    client = <<ClientCookie:8/binary>>, server = <<ServerCookie/binary>>
+}) when
+    8 =< byte_size(ServerCookie), byte_size(ServerCookie) =< 32
+->
+    {?DNS_EOPTCODE_COOKIE, <<ClientCookie/binary, ServerCookie/binary>>};
+do_encode_optrrdata(#dns_opt_cookie{}) ->
+    erlang:error(bad_cookie);
```

Comment:
This is the first time we are throwing an exception during the decoding of the opt RRs. This will start throwing exceptions and likely result in SERVFAIL. If we want to return anything other than SERVFAIL (FORMERR) we would need to handle the exception.